hpx/runtime/parcelset/parcelport.hpp

```diff
@@ -370,11 +350,10 @@ namespace hpx { namespace parcelset
         /// mutex for all of the member data
         mutable lcos::local::spinlock mtx_;
 
-        /// The handler for all incoming requests.
-        server::parcelport_queue parcels_;
+        hpx::applier::applier *applier_;
 
         /// The cache for pending parcels
-        typedef std::pair<std::vector<parcel>, std::vector<write_handler_type> >
+        typedef std::pair<std::deque<parcel>, std::deque<write_handler_type> >
```

Comment:
Frankly I have not fully understood your changes here. We didn't need `pop_front()` before, so why do we need it now?